src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -324,6 +324,20 @@ def getTask(self, taskName):
             return None
         return WMTaskHelper(task)
 
+    def getTaskByName(self, taskName):
+        """
+        _getTaskByName_
+
+        Retrieve a task with the given name in the whole workflow tree.
+        """
+        for t in self.taskIterator():
+            if t.name() == taskName:
+                return t
+            for x in t.taskIterator():
```

Comment:
Actually taskIterator is ok. Unlike workload.taskIterator() which only checks the top level task, task.taskIterator checks all the descendants. And childTaskIterator checks only first generation children.
Technically I could just remove

```
if t.name() == taskName:
    return t
```